Source/OCMock/NSInvocation+OCMAdditions.m

```diff
@@ -80,7 +80,16 @@ - (void)retainObjectArgumentsExcludingObject:(id)objectToExclude
     for(NSUInteger index = 2; index < numberOfArguments; index++)
     {
         const char *argumentType = [[self methodSignature] getArgumentTypeAtIndex:index];
-        if(OCMIsObjectType(argumentType))
+        BOOL isObjectType = OCMIsObjectType(argumentType);
+        if ([indexes containsIndex:index])
```

Comment:
Given the proposed implementation in the mock itself, the `indexes` argument here can be `nil`. Of course, the if statement still works in that case, but I must say it did confuse me a bit.

Reply:
Fixed up here as well I think